### PR TITLE
Feature/task 4

### PR DESCRIPTION
**Pull request by milan009**

### CODE REVIEW

File: onboarding-task/src/containers/AddItem.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+import { createItem } from '../actions/actionCreators';
+import { AddItem as AddItemComponent, IAddItemCallbackProps } from '../components/AddItem';
+import { IStore } from '../interfaces/IStore';
+
+const mapDispatchToProps = (dispatch: Dispatch<IStore>): IAddItemCallbackProps => ({
+  onAddItem: (text: string) => dispatch(createItem(text)),
+});
+
+const AddItem: React.ComponentClass<{}> = connect(null, mapDispatchToProps)(AddItemComponent);
```
Comment (Suzii) [inline onboarding-task/src/containers/AddItem.tsx]: Please, update your type definitions for react. The ones you operate with are dependant on TypeScript 2.2. In your app you are using TS 2.4 which already allows default generic parameters. Once you update, `<{}>` should be redundant.

File: onboarding-task/src/containers/List.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import { ItemsDataMap } from '../reducers/list/itemsReducer';
+import { IStore } from '../interfaces/IStore';
+
+const getIdsMemoized = memoize((items: ItemsDataMap) => items.keySeq());
+
+const mapStateToProps = (state: IStore) => ({
+  itemIds: getIdsMemoized(state.list.itemsById),
+});
+
+const List = connect(mapStateToProps)(ListComponent);
```
Comment (Suzii) [inline onboarding-task/src/containers/List.tsx]: Forgotten type for `List`.

File: onboarding-task/src/components/List.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+import { AddItem } from '../containers/AddItem';
+import { ListItem } from '../containers/ListItem';
+import { emptyUuid } from '../utils/constants';
+
+export interface IListDataProps {
+  itemIds: Seq.Indexed<string>;
+}
+
+const List: React.StatelessComponent<IListDataProps> = (props: IListDataProps) => {
```
Comment (Suzii) [inline onboarding-task/src/components/List.tsx]: You do not need to explicitely type props parameter, if it is specified in the generic :)

File: onboarding-task/package.json [hunk 4]
```diff
@@ -46,14 +50,16 @@
...
-    "typescript": "^2.1.5",
+    "typescript": "^2.4.2",
     "url-loader": "0.5.7",
     "webpack": "1.14.0",
     "webpack-dev-server": "1.16.2",
     "webpack-manifest-plugin": "1.1.0",
     "whatwg-fetch": "1.0.0"
   },
   "dependencies": {
+    "@types/classnames": "^2.2.0",
```
Comment (milan009) [inline onboarding-task/package.json]: Regarding the Immutable types, when looking for them, I've stumbled upon [this package](https://www.npmjs.com/package/@types/immutable). In its description it is mentioned that the package is a stub, and that the corresponding d.ts files are shipped with Immutable. Therefore, I did not install it separately :)

File: onboarding-task/src/actions/actionCreators.ts [hunk 5]
```diff
@@ -9,7 +9,7 @@ import {
   ITEM_MAKE_EDITABLE,
 } from './actionTypes';
 
-export const createItemFactory = (idGenerator: () => string) => (
+export const createItemFactory = (idGenerator: () => string): (text: string) => IAction => (
   (text: string) => ({
```
Comment (Suzii) [inline onboarding-task/src/actions/actionCreators.ts]: Inner lambda is sad because it didn't get its return type :sob:

**Pull request merged**